store/src/main/java/com/dropbox/android/external/store4/Store.kt

```diff
@@ -6,13 +6,40 @@ import kotlinx.coroutines.flow.filterNot
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.transform
 
-// possible replacement for [Store] as an internal only representation
-// if this class becomes public, should probaly be named IntermediateStore to distingush from
-// Store and also clarify that it still needs to be built/open? (how do we ensure?)
+/**
+ * A Store is responsible for managing a particular data request.
+ *
+ * When you create an implementation of a Store, you provide it with a Fetcher, a function that defines how data will be fetched over network.
+ *
+ * You can also define how your Store will cache data in-memory and on-disk. See [StoreBuilder] for full configuration
+ *
+ * Example usage:
+ *
+ * val store = StoreBuilder
+ *  .fromNonFlow<Pair<String, RedditConfig>, List<Post>> { (query, config) ->
+ *    provideRetrofit().fetchSubreddit(query, config.limit).data.children.map(::toPosts)
+ *   }
+ *  .persister(reader = { (query, _) -> db.postDao().loadPosts(query) },
+ *             writer = { (query, _), posts -> db.postDao().insertPosts(query, posts) },
+ *             delete = { (query, _) -> db.postDao().clearFeed(query) })
+ *  .build()
+ *  //single shot response
+ *  viewModelScope.launch {
```

Comment:
`viewModelScope` is an androidy concept. maybe `myCoroutineScope`?

Reply:
That was intentional, figured I want to steer folks on android to do the right thing